rustls/src/hash_hs.rs

```diff
@@ -98,10 +98,16 @@ impl HandshakeHash {
 
     /// Hash/buffer a handshake message.
     pub(crate) fn add_message(&mut self, m: &Message<'_>) -> &mut Self {
-        if let MessagePayload::Handshake { encoded, .. } = &m.payload {
-            self.update_raw(encoded.bytes());
+        match &m.payload {
+            MessagePayload::Handshake { encoded, .. } => self.update_raw(encoded.bytes()),
+            MessagePayload::HandshakeFlight(payload) => self.update_raw(payload.bytes()),
+            _ => self,
         }
-        self
+    }
+
+    /// Hash/buffer an encoded handshake message.
+    pub(crate) fn add(&mut self, bytes: &[u8]) {
```

Comment:
I'm chewing on the decision to add this wrapper instead of promoting `update_raw()`'s visibility. It feels like the main reason is to expose a name that's a better match to the new purpose of adding a handshake message that's already been encoded as an alternative to `add_message()`.

WDYT about renaming `update_raw()` to `add_raw()` instead?